objcgen/objcgenerator.cs

```diff
@@ -796,8 +796,124 @@ protected override void Generate (ProcessedMethod method)
+			implementation.WriteLine ();
+			implementation.WriteLine ("for (__residx = 0; __residx < __resarrlength; __residx++) {");
+			implementation.Indent++;
+
+			switch (typecode) {
+			case TypeCode.Boolean:
+				implementation.WriteLine ("bool __resarrval = mono_array_get (__resarr, bool, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithBool:__resarrval ? YES : NO];");
+				break;
+			case TypeCode.Char:
+				implementation.WriteLine ("unsigned short __resarrval = mono_array_get (__resarr, unsigned short, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithUnsignedShort:__resarrval];");
+				break;
+			case TypeCode.Double:
+				implementation.WriteLine ("double __resarrval = mono_array_get (__resarr, double, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithDouble:__resarrval];");
+				break;
+			case TypeCode.Single:
+				implementation.WriteLine ("float __resarrval = mono_array_get (__resarr, float, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithFloat:__resarrval];");
+				break;
+			case TypeCode.SByte:
+				implementation.WriteLine ("signed char __resarrval = mono_array_get (__resarr, signed char, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithChar:__resarrval];");
+				break;
+			case TypeCode.Int16:
+				implementation.WriteLine ("short __resarrval = mono_array_get (__resarr, short, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithShort:__resarrval];");
+				break;
+			case TypeCode.Int32:
+				implementation.WriteLine ("int __resarrval = mono_array_get (__resarr, int, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithInt:__resarrval];");
+				break;
+			case TypeCode.Int64:
+				implementation.WriteLine ("long long __resarrval = mono_array_get (__resarr, long long, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithLongLong:__resarrval];");
+				break;
+			case TypeCode.UInt16:
+				implementation.WriteLine ("unsigned short __resarrval = mono_array_get (__resarr, unsigned short, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithUnsignedShort:__resarrval];");
+				break;
+			case TypeCode.UInt32:
+				implementation.WriteLine ("unsigned int __resarrval = mono_array_get (__resarr, unsigned int, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithUnsignedInt:__resarrval];");
+				break;
+			case TypeCode.UInt64:
+				implementation.WriteLine ("unsigned long long __resarrval = mono_array_get (__resarr, unsigned long long, __residx);");
+				implementation.WriteLine ("__resobj = [NSNumber numberWithUnsignedLongLong:__resarrval];");
+				break;
+			case TypeCode.Byte:
+				implementation.WriteLine ("unsigned char __resarrval = mono_array_get (__resarr, unsigned char, __residx);");
+				implementation.WriteLine ("[__resobj appendBytes:&__resarrval length:1];");
```

Comment:
That will need to be optimized (byte arrays can be very large). You should be able to skip the loop using `mono_array_addr` and the array length, e.g.

```
[__resobj appendBytes:mono_array_addr (__resarr) length:__resarrlength];
```